### DISCUSSION TITLE
2023-05-30 개발일지

### DISCUSSION THREAD

**Discussion by Craftcode17** [Blue-club/pintos1#7] [개발일지]
2023-05-30

**Comment by carrotcookie** [comment]
특정 부분만 테스트 하고 싶을 때
위치는 threads/build. 코드 수정하면 해당위치에서 make clean, make 하고 아래 명령어 치면 됨.

pintos -v -k -T 60 -m 20   -- -q   run alarm-single < /dev/null 2> tests/threads/alarm-single.errors > tests/threads/alarm-single.output
perl -I../.. ../../tests/threads/alarm-single.ck tests/threads/alarm-single tests/threads/alarm-single.result

pintos -v -k -T 60 -m 20   -- -q   run alarm-multiple < /dev/null 2> tests/threads/alarm-multiple.errors > tests/threads/alarm-multiple.output
perl -I../.. ../../tests/threads/alarm-multiple.ck tests/threads/alarm-multiple tests/threads/alarm-multiple.result

pintos -v -k -T 60 -m 20   -- -q   run alarm-simultaneous < /dev/null 2> tests/threads/alarm-simultaneous.errors > tests/threads/alarm-simultaneous.output
perl -I../.. ../../tests/threads/alarm-simultaneous.ck tests/threads/alarm-simultaneous tests/threads/alarm-simultaneous.result

pintos -v -k -T 60 -m 20   -- -q   run alarm-priority < /dev/null 2> tests/threads/alarm-priority.errors > tests/threads/alarm-priority.output
perl -I../.. ../../tests/threads/alarm-priority.ck tests/threads/alarm-priority tests/threads/alarm-priority.result

pintos -v -k -T 60 -m 20   -- -q   run alarm-zero < /dev/null 2> tests/threads/alarm-zero.errors > tests/threads/alarm-zero.output
perl -I../.. ../../tests/threads/alarm-zero.ck tests/threads/alarm-zero tests/threads/alarm-zero.result

pintos -v -k -T 60 -m 20   -- -q   run alarm-negative < /dev/null 2> tests/threads/alarm-negative.errors > tests/threads/alarm-negative.output
perl -I../.. ../../tests/threads/alarm-negative.ck tests/threads/alarm-negative tests/threads/alarm-negative.result

pintos -v -k -T 60 -m 20   -- -q   run priority-change < /dev/null 2> tests/threads/priority-change.errors > tests/threads/priority-change.output
perl -I../.. ../../tests/threads/priority-change.ck tests/threads/priority-change tests/threads/priority-change.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-one < /dev/null 2> tests/threads/priority-donate-one.errors > tests/threads/priority-donate-one.output
perl -I../.. ../../tests/threads/priority-donate-one.ck tests/threads/priority-donate-one tests/threads/priority-donate-one.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-multiple < /dev/null 2> tests/threads/priority-donate-multiple.errors > tests/threads/priority-donate-multiple.output
perl -I../.. ../../tests/threads/priority-donate-multiple.ck tests/threads/priority-donate-multiple tests/threads/priority-donate-multiple.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-multiple2 < /dev/null 2> tests/threads/priority-donate-multiple2.errors > tests/threads/priority-donate-multiple2.output
perl -I../.. ../../tests/threads/priority-donate-multiple2.ck tests/threads/priority-donate-multiple2 tests/threads/priority-donate-multiple2.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-nest < /dev/null 2> tests/threads/priority-donate-nest.errors > tests/threads/priority-donate-nest.output
perl -I../.. ../../tests/threads/priority-donate-nest.ck tests/threads/priority-donate-nest tests/threads/priority-donate-nest.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-sema < /dev/null 2> tests/threads/priority-donate-sema.errors > tests/threads/priority-donate-sema.output
perl -I../.. ../../tests/threads/priority-donate-sema.ck tests/threads/priority-donate-sema tests/threads/priority-donate-sema.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-lower < /dev/null 2> tests/threads/priority-donate-lower.errors > tests/threads/priority-donate-lower.output
perl -I../.. ../../tests/threads/priority-donate-lower.ck tests/threads/priority-donate-lower tests/threads/priority-donate-lower.result

pintos -v -k -T 60 -m 20   -- -q   run priority-fifo < /dev/null 2> tests/threads/priority-fifo.errors > tests/threads/priority-fifo.output
perl -I../.. ../../tests/threads/priority-fifo.ck tests/threads/priority-fifo tests/threads/priority-fifo.result

pintos -v -k -T 60 -m 20   -- -q   run priority-preempt < /dev/null 2> tests/threads/priority-preempt.errors > tests/threads/priority-preempt.output
perl -I../.. ../../tests/threads/priority-preempt.ck tests/threads/priority-preempt tests/threads/priority-preempt.result

pintos -v -k -T 60 -m 20   -- -q   run priority-sema < /dev/null 2> tests/threads/priority-sema.errors > tests/threads/priority-sema.output
perl -I../.. ../../tests/threads/priority-sema.ck tests/threads/priority-sema tests/threads/priority-sema.result

pintos -v -k -T 60 -m 20   -- -q   run priority-condvar < /dev/null 2> tests/threads/priority-condvar.errors > tests/threads/priority-condvar.output
perl -I../.. ../../tests/threads/priority-condvar.ck tests/threads/priority-condvar tests/threads/priority-condvar.result

pintos -v -k -T 60 -m 20   -- -q   run priority-donate-chain < /dev/null 2> tests/threads/priority-donate-chain.errors > tests/threads/priority-donate-chain.output
perl -I../.. ../../tests/threads/priority-donate-chain.ck tests/threads/priority-donate-chain tests/threads/priority-donate-chain.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-load-1 < /dev/null 2> tests/threads/mlfqs/mlfqs-load-1.errors > tests/threads/mlfqs/mlfqs-load-1.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-load-1.ck tests/threads/mlfqs/mlfqs-load-1 tests/threads/mlfqs/mlfqs-load-1.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-load-60 < /dev/null 2> tests/threads/mlfqs/mlfqs-load-60.errors > tests/threads/mlfqs/mlfqs-load-60.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-load-60.ck tests/threads/mlfqs/mlfqs-load-60 tests/threads/mlfqs/mlfqs-load-60.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-load-avg < /dev/null 2> tests/threads/mlfqs/mlfqs-load-avg.errors > tests/threads/mlfqs/mlfqs-load-avg.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-load-avg.ck tests/threads/mlfqs/mlfqs-load-avg tests/threads/mlfqs/mlfqs-load-avg.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-recent-1 < /dev/null 2> tests/threads/mlfqs/mlfqs-recent-1.errors > tests/threads/mlfqs/mlfqs-recent-1.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-recent-1.ck tests/threads/mlfqs/mlfqs-recent-1 tests/threads/mlfqs/mlfqs-recent-1.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-fair-2 < /dev/null 2> tests/threads/mlfqs/mlfqs-fair-2.errors > tests/threads/mlfqs/mlfqs-fair-2.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-fair-2.ck tests/threads/mlfqs/mlfqs-fair-2 tests/threads/mlfqs/mlfqs-fair-2.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-fair-20 < /dev/null 2> tests/threads/mlfqs/mlfqs-fair-20.errors > tests/threads/mlfqs/mlfqs-fair-20.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-fair-20.ck tests/threads/mlfqs/mlfqs-fair-20 tests/threads/mlfqs/mlfqs-fair-20.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-block < /dev/null 2> tests/threads/mlfqs/mlfqs-block.errors > tests/threads/mlfqs/mlfqs-block.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-block.ck tests/threads/mlfqs/mlfqs-block tests/threads/mlfqs/mlfqs-block.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-nice-2 < /dev/null 2> tests/threads/mlfqs/mlfqs-nice-2.errors > tests/threads/mlfqs/mlfqs-nice-2.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-nice-2.ck tests/threads/mlfqs/mlfqs-nice-2 tests/threads/mlfqs/mlfqs-nice-2.result

pintos -v -k -T 480 -m 20   -- -q  -mlfqs run mlfqs-nice-10 < /dev/null 2> tests/threads/mlfqs/mlfqs-nice-10.errors > tests/threads/mlfqs/mlfqs-nice-10.output
perl -I../.. ../../tests/threads/mlfqs/mlfqs-nice-10.ck tests/threads/mlfqs/mlfqs-nice-10 tests/threads/mlfqs/mlfqs-nice-10.result

**Comment by Riudiu** [comment]
- alarm-priority, apriority-preempt 테스트 케이스 통과
- priority-fifo, priority-sema, priority-condvar (Extra) 테스트 케이스 에러나는 문제...
- 다음날 donate 부분까지 할 수 있도록 노력할 예정입니다
